src/config.rs

```diff
@@ -86,9 +101,21 @@ impl Config {
         }
     }
 
-    pub async fn check(&self) -> Result<()> {
+    pub async fn check(&self, skip_chain_id_validation: bool) -> Result<()> {
```

Comment:
I also think that two arguments seems a bit of overengineering it. I would go for the flag --skip-chain-id-validation, because it seems to me more convenient to add it when calling from console, instead of every time modifying file/environment variable/having multiple defined files. Also, it doesn't seem to me like a configuration, where we have setup urls, poll secs and so on.

Therefore, I will go with only flag, please write something if you are more for leaving it as an option in file.